arbiter/drivers/onedrive.cpp

```diff
@@ -0,0 +1,325 @@
+#ifndef ARBITER_IS_AMALGAMATION
+#include <arbiter/drivers/onedrive.hpp>
+#include <arbiter/arbiter.hpp>
+#include <arbiter/drivers/fs.hpp>
+#include <arbiter/util/transforms.hpp>
+#include <arbiter/util/json.hpp>
+#include <chrono>
+#include <thread>
+#endif
+
+#include <vector>
+
+#ifdef ARBITER_OPENSSL
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <openssl/rsa.h>
+#include <openssl/sha.h>
+
+// See: https://www.openssl.org/docs/manmaster/man3/OPENSSL_VERSION_NUMBER.html
+#   if OPENSSL_VERSION_NUMBER >= 0x010100000
+#   define ARBITER_OPENSSL_ATLEAST_1_1
+#   endif
+
+#endif
+
+#ifdef ARBITER_CUSTOM_NAMESPACE
+namespace ARBITER_CUSTOM_NAMESPACE
+{
+#endif
+
+namespace arbiter
+{
+
+using namespace internal;
+
+namespace {
+
+
+const std::string hostUrl = "https://graph.microsoft.com/v1.0/me/drive/root:/";
+
+std::string getBaseEndpoint(const std::string path)
+{
+    return std::string(hostUrl + path);
+}
+
+std::string getBinaryEndpoint(const std::string path)
+{
+    return std::string(path + ":/content");
+}
+
+std::string getChildrenEndpoint(const std::string path)
+{
+    return std::string(path + ":/children");
+}
+
+std::string getRefreshUrl()
+{
+    return "https://login.microsoftonline.com/common/oauth2/v2.0/token";
+}
+
+std::vector<char> buildBody(const http::Query& query)
```

Comment:
To avoid duplicating logic I'd l rather this be something like:
```
    const std::string s(http::buildQueryString(query));
    return std::vector<char>(s.begin(), s.end());
```

With that function modified not to include the leading `?` like it does now, which is more accurate to its name anyway.